sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/util/BinaryData.java

```diff
@@ -18,18 +20,27 @@
 import java.io.InputStream;
 import java.io.UncheckedIOException;
 import java.nio.ByteBuffer;
-import java.nio.charset.Charset;
 import java.nio.charset.StandardCharsets;
 import java.util.Arrays;
 import java.util.Objects;
 
 /**
- * This class is an abstraction over many different ways that binary data can be represented. The {@link BinaryData}
- * can be created from {@link InputStream}, {@link Flux} of {@link ByteBuffer}, {@link String}, {@link Object} and byte
- * array. The data is collected from provided sources and stored into a byte array.
+ * This class is an abstraction over many different ways that binary data can be represented. The data represented by
+ * {@link BinaryData} is immutable. The {@link BinaryData} can be created from {@link InputStream}, {@link Flux} of
+ * {@link ByteBuffer}, {@link String}, {@link Object} and byte array. This type internally store given data in byte
```

Comment:
```suggestion
 * {@link ByteBuffer}, {@link String}, {@link Object}, or byte array. This type internally store given data in byte
```

We're listing all types we can create `BinaryData` from, should use an `or`.